signal-desktop-source: rename to signal-desktop

flokli opened the pull request:
This effectively migrates users of the `signal-desktop` attribute to the from-source build.

We confirmed the from source build is also able to parse state produced by the binary version (not just with pretty empty state, but also bigger states with more historical data).

Rename the binary variant to `signal-desktop-bin`, consistent with how this is handled in other places in nixpkgs. Also update the pname to reflect that.

The testing mentioned in https://github.com/NixOS/nixpkgs/pull/395301#issuecomment-2772940217 has happened.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

flokli committed:
signal-desktop-source: rename to signal-desktop

This effectively migrates users of the `signal-desktop` attribute to the
from-source build.

We confirmed the from source build is also able to parse state produced
by the binary version (not just with pretty empty state, but also bigger
states with more historical data).

Rename the binary variant to `signal-desktop-bin`, consistent with how
this is handled in other places in nixpkgs. Also update the pname to
reflect that.

marcin-serwin reviewed (dismissed):
Update script of `signal-desktop-bin`  also needs to be updated due to rename.

This is a breaking change for darwin users that still reference `signal-desktop` since it will not build for them, are we okay with that?

flokli commented:
> Update script of `signal-desktop-bin` also needs to be updated due to rename.

Thanks, will do.

> This is a breaking change for darwin users that still reference `signal-desktop` since it will not build for them, are we okay with that?

Ugh, the source build for darwin not working indeed makes this a bit annoying.

I don't know. I personally think it also makes more sense for Darwin users to explicitly reference to `signal-desktop-bin` instead of `signal-desktop` silently pulling in a binary version. I'll update the release notes, mentioning people on Darwin need to update their configs and refer to `signal-desktop-bin` there. Maybe this motivates some Darwin user to fix the source build there too ;-)

flokli commented:
I updated the release notes and the update script. It already detected one update, but that's out of scope for this PR.

I also rebased, hopefully getting us past some unrelated failing check in the  nixpkgs manual (typst).
